.github/workflows/trigger-downstream-flake-update.yml

```
@@ -0,0 +1,30 @@
+name: Trigger Downstream Flake Update on Push to Main
+
+on:
+  push:
+    branches:
+      - main # Or the branch you want to track
+
+jobs:
+  trigger-downstream-repo:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        include:
+          - repo: 'blocksense-network/infra'
+    steps:
+      - uses: tibdex/github-app-token@v2.1.0
+        id: generate-token
+        with:
+          app_id: ${{ secrets.NIX_FLAKE_UPDATE_PR_BOT_APP_ID }}
+          private_key: ${{ secrets.NIX_FLAKE_UPDATE_PR_BOT_APP_PRIVATE_KEY }}
+
+      - name: Trigger Downstream Pipeline
+        run: |
+          curl -L -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: token ${{ steps.generate-token.outputs.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28"  \
+            "https://api.github.com/repos/${{ matrix.repo }}/dispatches" \
+            -d '{"event_type":"flake-update"}'
```

Can you add additional information about the branch which triggered the workflow?

```suggestion
            -d '{
              "event_type": "flake-update",
              "client_payload":{"branch-trigger": "${{ github.ref_name }}"}
            }'
```